source/trigger/cdc.go

```diff
@@ -0,0 +1,353 @@
+	var err error
+
+	iterator := &CDC{
+		conn:           params.Conn,
+		position:       params.Position,
+		table:          params.Table,
+		orderingColumn: params.OrderingColumn,
+		conduit:        params.Conduit,
+		key:            params.Key,
+		columns:        append(params.Columns, colID, colOperation),
+		batchSize:      params.BatchSize,
+		tableSrv:       newTrackingTableService(),
+	}
+
+	err = iterator.loadRows(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("load cdc rows: %w", err)
+	}
+
+	// run clearing tracking table.
+	go iterator.clearTrackingTable(ctx)
+
+	return iterator, nil
+}
+
+// HasNext returns a bool indicating whether the iterator has the next record to return or not.
+func (iter *CDC) HasNext(ctx context.Context) (bool, error) {
+	if iter.rows != nil && iter.rows.Next() {
+		return true, nil
+	}
+
+	if err := iter.loadRows(ctx); err != nil {
+		return false, fmt.Errorf("load rows: %w", err)
+	}
+
+	return iter.rows.Next(), nil
+}
+
+// Next returns the next record.
+func (iter *CDC) Next(_ context.Context) (sdk.Record, error) {
+	values, err := iter.rows.Values()
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("scan row values: %w", err)
+	}
+
+	row := make(map[string]any, len(values))
+	for i := range values {
+		row[iter.columns[i]] = values[i]
+	}
+
+	operationType, ok := row[colOperation].(string)
+	if !ok {
+		return sdk.Record{}, errWrongTrackingOperationType
+	}
+
+	// set a new position into the variable,
+	// to avoid saving position into the struct until we marshal the position
+	position := &Position{
+		Mode: ModeCDC,
+		// set the value from columnTrackingID column of the tracking table
+		LastProcessedVal: row[colID],
+		CreatedAt:        iter.position.CreatedAt,
+	}
+
+	convertedPosition, err := position.marshal()
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("convert position %w", err)
+	}
+
+	key := make(sdk.StructuredData)
+	if iter.key != "" {
+		val, ok := row[iter.key]
+		if !ok {
+			return sdk.Record{}, fmt.Errorf("key column %q not found", iter.key)
+		}
+
+		key[iter.key] = val
+	}
+
+	// delete tracking columns
+	delete(row, colOperation)
+	delete(row, colID)
+
+	rowBytes, err := json.Marshal(row)
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("marshal row: %w", err)
+	}
+
+	iter.position = position
+
+	metadata := sdk.Metadata{
+		common.MetadataPostgresTable: iter.table,
+	}
+	metadata.SetCreatedAt(time.Now())
+
+	switch operationType {
+	case actionInsert:
+		return sdk.Util.Source.NewRecordCreate(
+			convertedPosition,
+			metadata,
+			key,
+			sdk.RawData(rowBytes),
+		), nil
+	case actionUpdate:
+		return sdk.Util.Source.NewRecordUpdate(
+			convertedPosition,
+			metadata,
+			key,
+			nil,
+			sdk.RawData(rowBytes),
+		), nil
+	case actionDelete:
+		return sdk.Util.Source.NewRecordDelete(
+			convertedPosition,
+			metadata,
+			key,
+		), nil
+	default:
+		return sdk.Record{}, errWrongTrackingOperationType
+	}
```

Comment:
I think it would be okay to shrink the calls, as the arguments are not that long.
```suggestion
	switch operationType {
	case actionInsert:
		return sdk.Util.Source.NewRecordCreate(
		    convertedPosition, metadata, key, sdk.RawData(rowBytes),
		), nil
		
	case actionUpdate:
		return sdk.Util.Source.NewRecordUpdate(
		    convertedPosition, metadata, key, nil,sdk.RawData(rowBytes),
		), nil
		
	case actionDelete:
		return sdk.Util.Source.NewRecordDelete(
		    convertedPosition, metadata, key,
		), nil
		
	default:
		return sdk.Record{}, errWrongTrackingOperationType
	}
```

Reply:
Yes, it looks much better 👍